### DISCUSSION TITLE
Alternative way to perform heavy tasks

### DISCUSSION THREAD

**Discussion by pfazzi** [brefphp/bref#809] [Support]
Since the lambda environment is not compatible with `fastcgi_finish_request`, how can I execute heavy tasks easily after the response has been sent?

I tried with something like that, where in shutdown I execute some async tasks, but APIGateway still reaches the timeout limit...
```php
$response = $app->handle($request);

(new Laminas\HttpHandlerRunner\Emitter\SapiEmitter())->emit($response);

$app->shutdown();
```

**Comment by georgeboot** [comment]
Have you considered sending these tasks to a queue (for example, SQS) and processing them asynchronously?

**Comment by mnapoli** [comment]
Hi, just tried migrating this issue to a support discussion! It seems to have worked 😄 

And yes, SQS is a better pattern here, even though a bit more work.

@pfazzi you cannot execute tasks after emitting the response on Lambda. It seems like you can (if you read the code it does), but you can't. Lambda is frozen afterwards.

**Comment by pfazzi** [comment]
Yes, I totally agree with you guys, sending the task to a queue is maybe the best pattern. But you know, it costs more effort. In this case we can't spend more time on this feature, so we decided to go this way. Anyway I solved in another way my issue. Thanks anyway!
